### PR TITLE
tests/api: Make cleanup best effort

### PR DESCRIPTION
**Pull request by jrusz**
We should not fail on missing variables or errors during cleanup. The
test can fail in any stage and the cleanup should be run full each time.

Current example where this causes problem can be seen here: https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1990711774 
The test fails before the `S3_URL` is known and the cleanup function never reaches the part where background processes are killed and Gitlab CI gets stuck when there are processes running in background. This should ensure that the function is always ran in full.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: test/cases/api.sh [hunk 1]
```diff
@@ -194,12 +194,10 @@ function cleanupAWS() {
   AWS_SNAPSHOT_ID="${AWS_SNAPSHOT_ID:-}"
 
   if [ -n "$AWS_CMD" ]; then
-    set +e
```
Comment (Gundersanne) [inline test/cases/api.sh:197]: This will do the opposite of what you want I think? `set -e` means fast-failing.

**Review by Gundersanne — APPROVED** [review]

File: test/cases/api.sh (hunk 1 above)
Comment (jrusz) [reply test/cases/api.sh:197]: I've removed it from the individual functions and put it to the function calling these functions.

Comment (Gundersanne) [reply test/cases/api.sh:197]: Yup only noticed after, looks good :)

**Review by achilleas-k — APPROVED** [review]

**Comment by jrusz** [comment]
It works, it fails instead of getting stuck :) https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1992828358

**Comment by jrusz** [comment]
I'm afraid this will required admin merge again @Gundersanne ... The failed jobs are related to the ostree issues, this PR aims to prevent the `api.sh edge-commit` test from getting stuck which will be beneficial to the CI.

**Pull request merged**